handlers/coffee.go

```diff
@@ -35,3 +37,39 @@ func (c *Coffee) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
 
 	rw.Write(d)
 }
+
+// CreateCoffee creates a new coffee
+func (c *Coffee) CreateCoffee(_ int, rw http.ResponseWriter, r *http.Request) {
+	c.log.Info("Handle Coffee | CreateCoffee")
+
+	body := model.Coffee{}
+
+	err := json.NewDecoder(r.Body).Decode(&body)
+	if err != nil {
+		c.log.Error("Unable to decode JSON", "error", err)
+		http.Error(rw, "Unable to parse request body", http.StatusInternalServerError)
+		return
+	}
+
+	coffees, err := c.con.GetProductsByName(body.Name)
+	if len(coffees) > 0 {
+		c.log.Error("A coffee with the same name already exists", "error")
+		http.Error(rw, "A coffee with the same name already exists", http.StatusBadRequest)
+		return
+	}
```

Comment:
I think it's better to move this logic into `CreateCoffee` because you should check if the coffee exists before creating any coffee.

This check shouldn't be unique only to this handler.

Reply:
yeah agreed